src/pages/workspace/NewWorkspacePage.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import {View, Text} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import ONYXKEYS from '../../ONYXKEYS';
+import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import Navigation from '../../libs/Navigation/Navigation';
+import Permissions from '../../libs/Permissions';
+import styles from '../../styles/styles';
+import WorkspaceDefaultAvatar from '../../../assets/images/workspace-default-avatar.svg';
+import TextInputWithLabel from '../../components/TextInputWithLabel';
+import Button from '../../components/Button';
+import AttachmentPicker from '../../components/AttachmentPicker';
+import Icon from '../../components/Icon';
+import {DownArrow, Upload} from '../../components/Icon/Expensicons';
+import CreateMenu from '../../components/CreateMenu';
+import Switch from '../../components/Switch';
+import compose from '../../libs/compose';
+
+
+const propTypes = {
+    /** List of betas */
+    betas: PropTypes.arrayOf(PropTypes.string).isRequired,
+
+    ...withLocalizePropTypes,
+};
+
+class NewWorkspacePage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isEditPhotoMenuVisible: false,
+            name: '',
+            requestCall: false,
+        };
+
+        this.createMenuItems = this.createMenuItems.bind(this);
+    }
+
+    /**
+     * Create menu items list for avatar menu
+     *
+     * @param {Function} openPicker
+     * @returns {Array}
+     */
+    createMenuItems(openPicker) {
+        const menuItems = [
+            {
+                icon: Upload,
+                text: this.props.translate('workspace.new.uploadPhoto'),
+                onSelected: () => {
+                    openPicker({
+                        onPicked: () => {
+                            // TODO: connect with setWorkspaceAvatar function
```

Comment:
@HorusGoul are you up for creating a follow up issue to track the changes we need to make to "complete" this?

Reply:
Sure!